File: domain/src/main/java/domain/Vignette.java
package domain;

import domain.enums.VehicleType;
import domain.enums.VignetteType;

import javax.persistence.*;
import java.io.Serializable;

/**
 * Created by Michel on 27-2-2018.
 */
@Entity
public class Vignette implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Enumerated
    private VehicleType vehicleType;
    @Enumerated
    private VignetteType vignetteType;
    @Column
    private double price;

    public Vignette(){}

    public Vignette(VehicleType vehicleType, VignetteType vignetteType, double price){
        this.vehicleType = vehicleType;
        this.vignetteType = vignetteType;
        this.price = price;
    }

    public void setId(Long id){ this.id = id; }

    public Long getId() { return this.id; }

    public void setVehicleType(VehicleType vehicleType){
        this.vehicleType = vehicleType;
    }

    public VehicleType getVehicleType(){
        return this.vehicleType;
    }

    public void setVignetteType(VignetteType vignetteType){
        this.vignetteType = vignetteType;
    }

    public VignetteType getVignetteType(){
        return this.vignetteType;
    }

    public void setPrice(double price){
        this.price = price;
    }

    public double getPrice(){
        return this.price;
    }

}

File: src/main/java/domain/Vignette.kt
package domain

import domain.enums.VehicleType
import domain.enums.VignetteType
import java.util.UUID
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.EnumType
import javax.persistence.Enumerated
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "vignette")
data class Vignette(
    @Enumerated(EnumType.STRING)
    var vehicleType: VehicleType,
    @Enumerated(EnumType.STRING)
    var vignetteType: VignetteType,
    var price: Double
) {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    val id: Long = 0

    @Column(unique = true)
    var uuid: String = UUID.randomUUID().toString()
}
